server/client.go

```diff
@@ -3225,33 +3224,35 @@ func (c *client) Account() *Account {
 // prunePerAccountCache will prune off a random number of cache entries.
 func (c *client) prunePerAccountCache() {
 	n := 0
-	if time.Now().UnixNano()-c.in.losc >= int64(orphanSubsCheckInterval) {
+	now := time.Now().Unix()
+	if now-c.in.losc >= orphanSubsCheckInterval {
 		for cacheKey, pac := range c.in.pacache {
-			canRemove := true
+			remove := false
 			for _, sub := range pac.results.psubs {
-				if !sub.isClosed() {
-					canRemove = false
+				if sub.isClosed() {
+					remove = true
```

Comment:
I would just delete here and continue instead of tracking the remove boolean etc.

Reply:
there are several level of loops, so continue would just move to next psub (or qsub). What you want is continue the outer loop. I could use labels and gotos, but the code to remove (delete, n++) will have to be copied 3 times. Is that ok?

Reply:
ok, goto label REMOVE at the end that removes and a continue before that label.. will submit and tell me if that's what you had in mind.